ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -788,6 +833,286 @@ private void migrateToPropertyStore() throws InterruptedException {
     migrationComplete.await();
   }
 
+  /**
+   * Convert resources from semi-auto to full-auto.
+   * @param commaSeparatedResources the comma-separated list of resources that needs to be migrated
+   */
+  public void migrateToFullAuto(String commaSeparatedResources) throws InterruptedException {
+    CountDownLatch migrationComplete = new CountDownLatch(adminForDc.size());
+    // different DCs can be migrated in parallel
+    adminForDc.forEach((dcName, helixAdmin) -> Utils.newThread(() -> {
+      try {
+        logger.info("Starting resource migration to full-auto in {}", dcName);
+
+        // Get property store
+        ClusterMapConfig config = getClusterMapConfig(clusterName, dcName, null);
+        String zkConnectStr = dataCenterToZkAddress.get(dcName).getZkConnectStrs().get(0);
```

Comment:
before we do anything, we should verify that each resource is full_auto compatible, meaning, one host can't have partitions out from multiple resources.

Reply:
Sure. Added verification to see if resources are full_auto compatible